feat: added option to add card backs and changed order of cards

michaelkryukov opened the pull request:
I recently encountered challenges when trying to print two-sided cards that are easy to cut and glue. The primary issues were:

1. Difficulty in including card backs in the print.
2. No option to place the two sides of the card next to each other for easy alignment.

To address these, I made several changes to the codebase, resulting in a convenient card printing, cutting and gluing process.

<details>
<summary>The workflow has been tested and works great</summary>
  
![telegram-cloud-photo-size-2-5244811185256260409-y](https://github.com/user-attachments/assets/92d0df7b-3754-4c6b-ba69-943c467ee69a)
![telegram-cloud-photo-size-2-5249394898908611735-y](https://github.com/user-attachments/assets/be03ec31-c5b5-469e-8b26-f27e7dc1acfc)
![telegram-cloud-photo-size-2-5249394898908611730-y](https://github.com/user-attachments/assets/5df953f1-4b8e-4a13-9753-4b75a1ecf714)
![telegram-cloud-photo-size-2-5249394898908611734-y](https://github.com/user-attachments/assets/2dc98968-b679-41a3-bad2-37ff7e416135)
![telegram-cloud-photo-size-2-5249394898908611728-y](https://github.com/user-attachments/assets/5dca6171-5dcf-43fd-86c3-d36511f49d97)
  
</details>

NeAntonTsarev commented:
> I recently encountered challenges when trying to print two-sided cards that are easy to cut and glue. The primary issues were:
> 
> 1. Difficulty in including card backs in the print.
> 2. No option to place the two sides of the card next to each other for easy alignment.
> 
> To address these, I made several changes to the codebase, resulting in a convenient card printing, cutting and gluing process.
> 
> The workflow has been tested and works great

zalupa

haganbmj commented on src/views/ProxyPage.vue:
```diff
@@ -115,8 +122,11 @@
+            <template v-for="n in card.quantity" :key="n">
+                <img :src="card.selectedOption.url" v-if="shouldShowCard(card)">
+                <img :src="card.selectedOption.urlBack" v-if="shouldShowCard(card, 'back')">
+                <img src="card_back_highres.jpg" v-else-if="config.includeBacks">
```
This card back has a bigger border than the cropped images used for the fronts.
Your use case of actually gluing cards is pretty different from what I normally use this for, so don't know if that's something that would be an issue (or at least an annoyance) for you?

michaelkryukov replied:
I see. So `mtg-proxy-site` uses "border_crop" version of the images from scryfall. Looks like ideal solution for me is:

1. Change order of the cards when printing, so I can keep folding and gluing.
2. Add option to use "large" images instead of "border_crop" that that will use images like [that](https://cards.scryfall.io/normal/front/0/0/0003aa31-d42e-4972-9fb9-086aefe29a2c.jpg?1675536851). It's even [not that hard](https://codepen.io/michaelkrukov/pen/mdZERYd) to fill the corners.
3. Add option to add card backs when needed (with two card back images – one looking like "corder_crop" and other looking like "large").

That way current use cases are not really affected while I get functionality I want. I guess adding another two options is not ideal, but I have couple ideas how to tidy interface up I can try. I can split changes into different PRs, or do you think you wouldn't accept them?

michaelkryukov commented:
Changes added after the last review:

- Added option to use 'normal' images instead of 'border_crop' – this changed stored json files. I did update lorcana file too. 
- Added two versions of card back (with proper sizes and areas).
- Rearranged checkboxes on the page.
- Removed rounding for cards being previewed to make it closer to what will be printed.
- 
I liked the results and that everything that worked still does. On the other hand, this PR is now quite large. I would like to hear feedback on the changes - I can split this PR into smaller PRs with some changes (order, ability to include backdrops, ability to use "normal" images from Scryfall, placement of checkboxes).

<details>
<summary>Images of the interface and printed cards</summary>

<img width="1266" alt="Screenshot 2024-07-23 at 00 31 11" src="https://github.com/user-attachments/assets/b812c9e6-0eeb-4f8f-956b-c7387cadc9c0">

![telegram-cloud-photo-size-2-5258425673238830819-y](https://github.com/user-attachments/assets/0c26b7a3-05a1-45cf-8dba-404f1e8f697e)
  
</details>

haganbmj commented:
> Changes added after the last review:
> 
> * Added option to use 'normal' images instead of 'border_crop' – this changed stored json files. I did update lorcana file too.
> * Added two versions of card back (with proper sizes and areas).
> * Rearranged checkboxes on the page.
> * Removed rounding for cards being previewed to make it closer to what will be printed.
> 
> I liked the results and that everything that worked still does. On the other hand, this PR is now quite large. I would like to hear feedback on the changes and I can split this PR into smaller PRs with only some of the changes (order, ability to include card backs, rounded border, ability to use "normal" images from scryfall, placement of checkboxes).

Don't worry about the Lorcana files, probably just going to remove those since I don't have a great source for a card list or images.

haganbmj commented on download-scryfall-cards.mjs:
```diff
@@ -107,8 +107,14 @@ const stripped = cards.filter(card => {
-            front: `https://api.scryfall.com/cards/${card.set}/${card.collector_number}?format=image&version=border_crop&face=front`,
-            back: card.card_faces?.[1]?.image_uris ? `https://api.scryfall.com/cards/${card.set}/${card.collector_number}?format=image&version=border_crop&face=back` : undefined,
+            front: {
+                cropped: `https://api.scryfall.com/cards/${card.set}/${card.collector_number}?format=image&version=border_crop&face=front`,
+                normal: `https://api.scryfall.com/cards/${card.set}/${card.collector_number}?format=image&version=normal&face=front`,
```
These endpoints only differ by a query param, so maybe instead of storing the full url here it's easier to just modify that query param where it's used?

haganbmj replied:
I'm writing up a quick commit to maybe switch to that. Just realized that I don't have node installed on this computer though so gonna take me a few minutes longer than expected >.>

haganbmj replied:
Here's a trimmed down commit that _just_ tackles switching between the Border Crop and Normal images.

It doesn't
- Fix the card count.
- Re-add your feature to print non-dfc card backs.
- Update the print scale.
- Rearrange the configuration.

haganbmj replied:
Added another commit, takes care of configuring Card Backs and fixes the count behavior.

haganbmj replied:
![Screenshot 2024-07-24 at 10 31 54 PM](https://github.com/user-attachments/assets/8a4b8289-aef6-4a71-b9f8-805f485f409c)

Would something like this work for now? 
Just added "Actual" as a quick option that has the sizing you want.

haganbmj commented:
I'm going to merge #201 and close these two PRs. 
I added what I think was the last feature here: collating the cards differently.

A thing I was noticing while trying out the "normal" Scryfall images is that it has a Black background for white bordered cards and some newer cards (MH3) have square corners instead of rounded.
